cloudify_agent/api/plugins/installer.py

```diff
@@ -221,3 +262,89 @@ def extract_package_name(package_dir):
         cwd=package_dir
     ).std_out
     return plugin_name
+
+
+def get_plugin_id(plugin):
+    package_name = plugin.get('package_name')
+    package_version = plugin.get('package_version')
+    distribution = plugin.get('distribution')
+    distribution_version = plugin.get('distribution_version')
+    distribution_release = plugin.get('distribution_release')
+    supported_platform = plugin.get('supported_platform')
+
+    if not package_name:
+        return None
+
+    query_parameters = {
+        'package_name': package_name
+    }
+    if package_version:
+        query_parameters['package_version'] = package_version
+    if distribution:
+        query_parameters['distribution'] = distribution
+    if distribution_version:
+        query_parameters['distribution_version'] = distribution_version
+    if distribution_release:
+        query_parameters['distribution_release'] = distribution_release
+    if supported_platform:
+        query_parameters['supported_platform'] = supported_platform
+    client = get_rest_client()
+    plugins = client.plugins.list(**query_parameters)
+
+    if not supported_platform:
+        plugins = [p for p in plugins
+                   if p.supported_platform in ['any', os.name]]
+    if os.name != 'nt':
+        a_dist, _, a_dist_release = platform.linux_distribution(
+            full_distribution_name=False)
+        if not distribution:
+            plugins = [p for p in plugins
+                       if p.supported_platform == 'any' or
+                       p.distribution == a_dist]
+        if not distribution_release:
+            plugins = [p for p in plugins
+                       if p.supported_platform == 'any' or
+                       p.distribution_release == a_dist_release]
+
+    if not plugins:
+        return None
+
+    plugin_result = max(plugins,
+                        key=lambda plug: LooseVersion(plug.package_version))
+    return plugin_result.id
+
+
+def get_plugin_source(plugin, blueprint_id=None):
+
+    source = plugin.get('source') or ''
+    if not source:
+        return None
+    source = source.strip()
+
+    # validate source url
+    if '://' in source:
+        split = source.split('://')
+        schema = split[0]
+        if schema not in ['http', 'https']:
```

Comment:
btw, I dunno how exactly it works with banks and the likes.. but I'm pretty sure we need to start supporting ftp for downloads :)